core/src/main/java/feign/ReflectiveFeign.java

```diff
@@ -196,7 +198,37 @@ public RequestTemplate create(Object[] argv) {
           }
         }
       }
-      return resolve(argv, mutable, varBuilder);
+
+      RequestTemplate template = resolve(argv, mutable, varBuilder);
+
+      // add query map parameters after initial resolve so that they take
+      // precedence over any predefined values
+      return addQueryMapQueryParameters(argv, template);
```

Comment:
not sure if this should be last method in resolve or not.. looks a bit random here.